docs/reference/asyncapi-rules.md

```diff
@@ -0,0 +1,325 @@
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+```
+
+### asyncapi2-operation-description
+
+Operation objects should have a description.
+
+**Recommended:** Yes
+
+### asyncapi2-operation-operationId
+
+This operation ID is essentially a reference for the operation, which can be used to visually suggest a connection to other operations. This is like some theoretical static HATEOAS-style referencing, but it's also used for the URL in some documentation systems.
+
+Make the value `lower-hyphen-case`, and try and think of a name for the action which does not relate to the HTTP message. Base it off the actual action being performed. `create-polygon`? `search-by-polygon`? `filter-companies`?
+
+**Recommended:** Yes
+
+### asyncapi2-parameter-description
+
+Parameter objects should have a `description`.
+
+**Recommended:** No
+
+### asyncapi2-payload-default
+
+`default` objects should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-payload-example
+
+`example` objects should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-payload-examples
+
+Values of the `examples` array should be valid against the `payload` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-payload-unsupported-schemaFormat
+
+Spectral currently only supports payload validation against the default AsyncAPI schema.
+
+This rule will notify you when a payload is decorated with a `schemaFormat` property. Spectral won't be able to validate it and potential issues in this payload will go undetected by the `asyncapi2-payload-*` rules.
+
+**Recommended:** Yes
+
+### asyncapi2-payload
+
+When `schemaFormat` is undefined, the `payload` object should be valid against the AsyncAPI 2 Schema Object definition.
+
+**Recommended:** Yes
+
+### asyncapi2-schema-default
+
+`default` objects should be valid against the `schema` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-schema-example
+
+`example` objects should be valid against the `schema` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-schema-examples
+
+Values of the `examples` array should be valid against the `schema` they decorate.
+
+**Recommended:** Yes
+
+### asyncapi2-schema
+
+Validate structure of AsyncAPI v2 specification.
+
+**Recommended:** Yes
+
+### asyncapi2-server-no-empty-variable
+
+Server URL variable declarations cannot be empty, ex.`gigantic-server.com/{}` is invalid.
+
+**Recommended:** Yes
+
+### asyncapi2-server-no-trailing-slash
+
+Server URL should not have a trailing slash.
+
+Some tooling forgets to strip trailing slashes off when it's joining the `servers.url` with `channels`, and you can get awkward URLs like `https://example.com/api//pets`. Best to just strip them off yourself.
+
+**Recommended:** Yes
+
+**Good Example**
+
+``` yaml
+servers:
+  - url: https://example.com
+  - url: https://example.com/api
```

Comment:
It would be great not to use HTTP as it's one of the less common protocols in messaging.

```suggestion
  - url: mqtt://example.com
  - url: mqtt://example.com/broker
```

Reply:
@fmvilas Fixed